Exception thrown when deleting an appointment after its patient/doctor was deleted first

When entering `delete-appt` after the patient or doctor for that appointment was deleted beforehand, an exception is thrown. Entering the command for the first time will do nothing. Entering the command again (ie. pressing the enter key again for a second time) will display "Invalid Unique ID, appointment does not exist.".

However, if the displayed list is refreshed (eg. entering `list-patient` or `list-doctor`), the appointment will no longer be displayed.

Steps to reproduce:

```
add-doctor n/John Doe p/98765432 e/johndoe@example.com a/123 Clementi Ave 3 s/Orthopedics
add-patient n/Jane Doe p/91234567 e/janedoe@example.com a/456 Clementi Ave 3 d/01-05-1990 g/F t/friend
add-appt pn/Jane Doe dn/John Doe d/23-04-1987 t/1100
delete 2
delete-appt <ID of appointment>
```

Part of the exception thrown:

```
Caused by: seedu.address.model.person.exceptions.PersonNotFoundException
        at seedu.address.model.person.UniquePersonList.setPerson(UniquePersonList.java:61)
        at seedu.address.model.AddressBook.setPerson(AddressBook.java:99)
        at seedu.address.model.ModelManager.setPerson(ModelManager.java:112)
        at seedu.address.logic.commands.DeleteAppointmentCommand.execute(DeleteAppointmentCommand.java:63)
        at seedu.address.logic.LogicManager.execute(LogicManager.java:51)
        at seedu.address.ui.MainWindow.executeCommand(MainWindow.java:177)
        at seedu.address.ui.CommandBox.handleCommandEntered(CommandBox.java:45)
```

--- a/src/main/java/seedu/address/logic/commands/DeleteCommand.java
+++ b/src/main/java/seedu/address/logic/commands/DeleteCommand.java
@@ -24,6 +24,8 @@ public class DeleteCommand extends Command {
             + "Example: " + COMMAND_WORD + " 1";
 
     public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_WITH_APPOINTMENTS =
+            "Unable to delete person as they still have appointments. Delete them first.";
 
     private final Index targetIndex;
 
@@ -41,6 +43,11 @@ public class DeleteCommand extends Command {
         }
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+
+        if (!personToDelete.getAppointments().isEmpty()) {
+            throw new CommandException(MESSAGE_DELETE_PERSON_WITH_APPOINTMENTS);
+        }
+
         model.deletePerson(personToDelete);
         return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, Messages.format(personToDelete)));
     }

